src/app/games/yinsh/YinshMinimax.ts

```diff
@@ -0,0 +1,111 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { HexaDirection } from 'src/app/jscaip/HexaDirection';
+import { Minimax } from 'src/app/jscaip/Minimax';
+import { NodeUnheritance } from 'src/app/jscaip/NodeUnheritance';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus } from 'src/app/jscaip/Rules';
+import { Combinatorics } from 'src/app/utils/Combinatorics';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { GipfMinimax } from '../gipf/GipfMinimax';
+import { GipfCapture } from '../gipf/GipfMove';
+import { YinshGameState } from './YinshGameState';
+import { YinshLegalityStatus } from './YinshLegalityStatus';
+import { YinshCapture, YinshMove } from './YinshMove';
+import { YinshPiece } from './YinshPiece';
+import { YinshNode, YinshRules } from './YinshRules';
+
+export class YinshMinimax extends Minimax<YinshMove, YinshGameState, YinshLegalityStatus> {
+    public getBoardValue(node: YinshNode): NodeUnheritance {
+        const status: GameStatus = this.ruler.getGameStatus(node);
+        if (status.isEndGame) {
+            return NodeUnheritance.fromWinner(status.winner);
+        } else {
+            return new NodeUnheritance(node.gamePartSlice.sideRings[0] * Player.ZERO.getScoreModifier() +
+                node.gamePartSlice.sideRings[1] * Player.ONE.getScoreModifier());
+        }
+    }
+    public getListMoves(node: YinshNode): YinshMove[] {
+        const moves: YinshMove[] = [];
+        const state: YinshGameState = node.gamePartSlice;
+
+        if (state.isInitialPlacementPhase()) {
+            state.hexaBoard.forEachCoord((coord: Coord, content: YinshPiece): void => {
```

Comment:
Askip p.d.v performance les .forEach c'est vachement moins bon que les for normal, et en lisibilité c'est aussi un monde différent

Reply:
Donc ici, plutôt que d'appeler le callback sur chaque élément, il faudrait d'abord construire une liste de toutes les cases, la retourner, puis `for`er dessus ? Niveau perf pas convaincu, cependant agreed niveau lisibilité !